From 260470de3e65ec72d7f206c4ce21c621bfea53d9 Mon Sep 17 00:00:00 2001
From: Andrew Randrianasulu <randrianasulu@gmail.com>
Date: Tue, 12 Sep 2023 17:37:14 +0300
Subject: [PATCH] Possible fix for group move hang

---
 cinelerra-5.1/cinelerra/edl.C | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cinelerra-5.1/cinelerra/edl.C b/cinelerra-5.1/cinelerra/edl.C
index 0780ac72..f1886dde 100644
--- a/cinelerra-5.1/cinelerra/edl.C
+++ b/cinelerra-5.1/cinelerra/edl.C
@@ -929,9 +929,12 @@ void EDL::delete_edit_labels(ArrayList<Edit*> *edits, int collapse)
 	ArrayList<Range> regions;
 	get_edit_regions(edits, regions);
 	int n = regions.size(), k = n;
+	//printf("delete_edit_label: k= %i\n", k);
+	if (k>0) {
 	while( --k >= 0 ) {
 		Range &range = regions[k];
 		labels->clear(range.start, range.end, collapse);
+	    }
 	}
 }
 
-- 
2.35.7

