diff --git a/cinelerra-5.1/cinelerra/theme.C b/cinelerra-5.1/cinelerra/theme.C
index d89c789f..ab22ec8e 100644
--- a/cinelerra-5.1/cinelerra/theme.C
+++ b/cinelerra-5.1/cinelerra/theme.C
@@ -274,6 +274,7 @@ void Theme::build_menus()
 	aspect_ratios.append(new BC_ListBoxItem("3:2"));
 	aspect_ratios.append(new BC_ListBoxItem("4:3"));
 	aspect_ratios.append(new BC_ListBoxItem("16:9"));
+	aspect_ratios.append(new BC_ListBoxItem("16:10"));
 	aspect_ratios.append(new BC_ListBoxItem("2.10:1"));
 	aspect_ratios.append(new BC_ListBoxItem("2.20:1"));
 	aspect_ratios.append(new BC_ListBoxItem("2.25:1"));
@@ -283,9 +284,11 @@ void Theme::build_menus()
 
 	frame_sizes.append(new BC_ListBoxItem("128x96     "));
 	frame_sizes.append(new BC_ListBoxItem("160x120    "));
+	frame_sizes.append(new BC_ListBoxItem("240x160   HQVGA"));
 	frame_sizes.append(new BC_ListBoxItem("320x240    "));
 	frame_sizes.append(new BC_ListBoxItem("360x240    "));
-	frame_sizes.append(new BC_ListBoxItem("400x300    "));
+	frame_sizes.append(new BC_ListBoxItem("384x288    CIF"));
+	frame_sizes.append(new BC_ListBoxItem("400x300    qSVGA"));
 	frame_sizes.append(new BC_ListBoxItem("640x360    nHD"));
 	frame_sizes.append(new BC_ListBoxItem("640x400    "));
 	frame_sizes.append(new BC_ListBoxItem("640x480    VGA"));
@@ -299,19 +302,32 @@ void Theme::build_menus()
 	frame_sizes.append(new BC_ListBoxItem("1024x576   "));
 	frame_sizes.append(new BC_ListBoxItem("1024x768   XGA"));
 	frame_sizes.append(new BC_ListBoxItem("1152x648   "));
+	frame_sizes.append(new BC_ListBoxItem("1152x768   "));
+	frame_sizes.append(new BC_ListBoxItem("1152x864  XGA+"));
 	frame_sizes.append(new BC_ListBoxItem("1280x720   HD"));
+	frame_sizes.append(new BC_ListBoxItem("1280x800   "));
 	frame_sizes.append(new BC_ListBoxItem("1280x1024  SXGA"));
 	frame_sizes.append(new BC_ListBoxItem("1366x768   WXGA"));
+	frame_sizes.append(new BC_ListBoxItem("1440x900   WXGA+"));
+	frame_sizes.append(new BC_ListBoxItem("1440x1080  HDV"));
 	frame_sizes.append(new BC_ListBoxItem("1600x900   HD+"));
 	frame_sizes.append(new BC_ListBoxItem("1600x1200  UXGA"));
 	frame_sizes.append(new BC_ListBoxItem("1920x1080  Full HD"));
+	frame_sizes.append(new BC_ListBoxItem("1920x1280  FHD+"));
+	frame_sizes.append(new BC_ListBoxItem("2048x1080  DCI 2K"));
 	frame_sizes.append(new BC_ListBoxItem("2048x1152  "));
+	frame_sizes.append(new BC_ListBoxItem("2048x1536   QXGA"));
+	frame_sizes.append(new BC_ListBoxItem("2160x1440   "));
 	frame_sizes.append(new BC_ListBoxItem("2304x1296  "));
-	frame_sizes.append(new BC_ListBoxItem("2560x1440  QHD"));
+	frame_sizes.append(new BC_ListBoxItem("2560x1080  UW-FHD"));
+	frame_sizes.append(new BC_ListBoxItem("2560x1440  WQHD"));
 	frame_sizes.append(new BC_ListBoxItem("2880x1620  "));
-	frame_sizes.append(new BC_ListBoxItem("3200x1800  QHD+"));
+	frame_sizes.append(new BC_ListBoxItem("3200x1800  WQXGA+"));
+	frame_sizes.append(new BC_ListBoxItem("3440x1440  Ultra-Wide QHD"));
 	frame_sizes.append(new BC_ListBoxItem("3520x1980  "));
+	frame_sizes.append(new BC_ListBoxItem("3840x1600  UW4K"));
 	frame_sizes.append(new BC_ListBoxItem("3840x2160  4K UHD"));
+	frame_sizes.append(new BC_ListBoxItem("4096x2160  DCI 4K"));
 	frame_sizes.append(new BC_ListBoxItem("4096x2304  Full 4K UHD"));
 	frame_sizes.append(new BC_ListBoxItem("4480x2520  "));
 	frame_sizes.append(new BC_ListBoxItem("5120x2880  5K UHD"));
