diff --git a/cinelerra-5.1/cinelerra/versioninfo.h b/cinelerra-5.1/cinelerra/versioninfo.h
index c100778..70ed482 100644
--- a/cinelerra-5.1/cinelerra/versioninfo.h
+++ b/cinelerra-5.1/cinelerra/versioninfo.h
@@ -3,9 +3,9 @@
 
 #define CINELERRA_VERSION "Infinity"
 #define REPOMAINTXT "git://git.cinelerra-gg.org/goodguy/cinelerra.git\n"
-#define COPYRIGHT_DATE "2018"
-#define COPYRIGHTTEXT1 "(c) 2006-2018 Heroine Virtual Ltd. by Adam Williams\n"
-#define COPYRIGHTTEXT2 "(c) 2007-2018 cin5 derivative by W.P. Morrow aka goodguy\n"
+#define COPYRIGHT_DATE "2019"
+#define COPYRIGHTTEXT1 "(c) 2006-2019 Heroine Virtual Ltd. by Adam Williams\n"
+#define COPYRIGHTTEXT2 "(c) 2007-2019 cin5 derivative by W.P. Morrow aka goodguy\n"
 #undef COMPILEDATE
 
 #endif
