--- cinelerra/main.C.orig	2017-03-14 19:23:23.000000000 +0300
+++ cinelerra/main.C	2017-05-19 17:55:31.769094152 +0300
@@ -75,6 +75,9 @@
 	Garbage::garbage = new Garbage;
 	EDL::id_lock = new Mutex("EDL::id_lock");
 
+	/* dri3 crashes if OpenGL rendering selected  with nouveau driver as of 19-05-2017 */
+	setenv("LIBGL_DRI3_DISABLE", "1", 1);
+
 	bindtextdomain (GETTEXT_PACKAGE, LOCALE_DIR);
 	textdomain (GETTEXT_PACKAGE);
 	setlocale (LC_MESSAGES, "");
