diff --git a/cinelerra-5.1/cinelerra/Makefile b/cinelerra-5.1/cinelerra/Makefile
index 81d1ace..e80f8f7 100644
--- a/cinelerra-5.1/cinelerra/Makefile
+++ b/cinelerra-5.1/cinelerra/Makefile
@@ -496,7 +496,7 @@ CFLAGS += -DUSE_ALPHA
 else
 
 LDFLAGS1 = -Wl,-export-dynamic -g
-LDFLAGS2 = $(LDFLAGS)
+LDFLAGS2 = $(LDFLAGS) -ltiff
 LINKER = $(CXX) -o $(OUTPUT)
 
 endif
diff --git a/cinelerra-5.1/cinelerra/ci b/cinelerra-5.1/cinelerra/ci
index bc98783..2285898 120000
--- a/cinelerra-5.1/cinelerra/ci
+++ b/cinelerra-5.1/cinelerra/ci
@@ -1 +1 @@
-../bin/cinelerra
\ No newline at end of file
+../bin/cin
\ No newline at end of file
diff --git a/cinelerra-5.1/cinelerra/commercials.h b/cinelerra-5.1/cinelerra/commercials.h
index b7b815c..01f4c27 100644
--- a/cinelerra-5.1/cinelerra/commercials.h
+++ b/cinelerra-5.1/cinelerra/commercials.h
@@ -1,3 +1,4 @@
+#ifdef __i386__
 #ifndef _COMMERCIALS_H_
 #define _COMMERCIALS_H_
 
@@ -202,3 +203,4 @@ public:
 
 
 #endif
+#endif /* i386 */
\ No newline at end of file
diff --git a/cinelerra-5.1/cinelerra/dbwindow.h b/cinelerra-5.1/cinelerra/dbwindow.h
index 24a40df..983c65b 100644
--- a/cinelerra-5.1/cinelerra/dbwindow.h
+++ b/cinelerra-5.1/cinelerra/dbwindow.h
@@ -1,3 +1,4 @@
+#ifdef __i386__
 #ifndef _DBWINDOW_H_
 #define _DBWINDOW_H_
 
@@ -296,3 +297,4 @@ public:
 
 
 #endif
+#endif /* i386 */
\ No newline at end of file
diff --git a/cinelerra-5.1/cinelerra/mediadb.h b/cinelerra-5.1/cinelerra/mediadb.h
index 9bb970e..eb39340 100644
--- a/cinelerra-5.1/cinelerra/mediadb.h
+++ b/cinelerra-5.1/cinelerra/mediadb.h
@@ -1,3 +1,4 @@
+#ifdef __i386__
 #ifndef _MEDIA_DB_H_
 #define _MEDIA_DB_H_
 
@@ -239,3 +240,4 @@ public:
 };
 
 #endif
+#endif /* i386 */
\ No newline at end of file
diff --git a/cinelerra-5.1/cinelerra/mwindow.C b/cinelerra-5.1/cinelerra/mwindow.C
index 04549fd..bff4d9b 100644
--- a/cinelerra-5.1/cinelerra/mwindow.C
+++ b/cinelerra-5.1/cinelerra/mwindow.C
@@ -284,7 +284,9 @@ MWindow::~MWindow()
 	delete render;          render = 0;
 	delete mixers_align;    mixers_align = 0;
 	commit_commercial();
+#ifdef HAVE_COMMERCIALS
 	if( commercials && !commercials->remove_user() ) commercials = 0;
+#endif
 	close_mixers();
 	if( speed_edl ) { speed_edl->remove_user();  speed_edl = 0; }
 // Save defaults for open plugins
diff --git a/cinelerra-5.1/configure.ac b/cinelerra-5.1/configure.ac
index d529a80..9f1b51a 100644
--- a/cinelerra-5.1/configure.ac
+++ b/cinelerra-5.1/configure.ac
@@ -319,8 +319,7 @@ PKG_3RD([libvorbis],[auto],
 
 PKG_3RD([mjpegtools],[yes],
   [mjpegtools-2.1.0],
-  [ utils/mmxsse/.libs/libmmxsse.a \
-    utils/.libs/libmjpegutils.a \
+  [ utils/.libs/libmjpegutils.a \
     lavtools/.libs/liblavfile.a \
     lavtools/.libs/liblavjpeg.a \
     mplex/.libs/libmplex2.a \
@@ -363,7 +362,7 @@ PKG_3RD([twolame],[auto],
   [ libtwolame/.libs/libtwolame.a ],
   [  libtwolame ])
 
-PKG_3RD([x264],[auto],
+PKG_3RD([x264],[no],
   [x264-snapshot-20191217-2245-stable],
   [ libx264.a ],
   [ . ])
diff --git a/cinelerra-5.1/db/tdb.h b/cinelerra-5.1/db/tdb.h
index 8ee8800..b337b90 100644
--- a/cinelerra-5.1/db/tdb.h
+++ b/cinelerra-5.1/db/tdb.h
@@ -1,3 +1,5 @@
+#ifdef __i386__
+
 #ifndef __DB_H__
 #define __DB_H__
 #include <cstring>
@@ -1514,3 +1516,5 @@ public:
 };
 
 #endif
+
+#endif /* i386 */
\ No newline at end of file
diff --git a/cinelerra-5.1/doc/cinelerra.html b/cinelerra-5.1/doc/cinelerra.html
index f7742b5..de7e921 100644
--- a/cinelerra-5.1/doc/cinelerra.html
+++ b/cinelerra-5.1/doc/cinelerra.html
@@ -1,6 +1,6 @@
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
 <html>
-<!-- Created by GNU Texinfo 6.0, http://www.gnu.org/software/texinfo/ -->
+<!-- Created by GNU Texinfo 6.1, http://www.gnu.org/software/texinfo/ -->
 <head>
 <title>SECRETS OF CINELERRA</title>
 
@@ -34,9 +34,8 @@ pre.smalldisplay {font-family: inherit; font-size: smaller}
 pre.smallexample {font-size: smaller}
 pre.smallformat {font-family: inherit; font-size: smaller}
 pre.smalllisp {font-size: smaller}
-span.nocodebreak {white-space: nowrap}
 span.nolinebreak {white-space: nowrap}
-span.roman {font-family: serif; font-weight: normal}
+span.roman {font-family: initial; font-weight: normal}
 span.sansserif {font-family: sans-serif; font-weight: normal}
 ul.no-bullet {list-style: none}
 -->
@@ -6682,6 +6681,7 @@ int x_accum, y_accum;
 </p>
 <div class="example">
 <pre class="example">
+
 	switch (code) {
      		case REL_X:
      			mousedev-&gt;packet.x_accum += value * DOWNSAMPLE_N;
@@ -6796,6 +6796,7 @@ the fastest file system is
 </p>
 <div class="example">
 <pre class="example">
+
 mke2fs -i 65536 -b 4096 my_device
 tune2fs -r0 -c10000 my_device
 
@@ -6808,6 +6809,7 @@ failures is
 </p>
 <div class="example">
 <pre class="example">
+
 mke2fs -j -i 65536 -b 4096 my_device
 tune2fs -r0 -c10000 my_device
 
@@ -9764,6 +9766,7 @@ variables to yield the current values.
 </p>
 <div class="example">
 <pre class="example">
+
 	this-&gt;parameter1 = (float)(prev.parameter1 * prev_scale + next.parameter1 * next_scale);
 	this-&gt;parameter2 = (float)(prev.parameter2 * prev_scale + next.parameter2 * next_scale);
 	this-&gt;parameter3 = (int)(prev.parameter3 * prev_scale + next.parameter3 * next_scale);
@@ -9829,6 +9832,7 @@ appropriate arguments.
 </p>
 <div class="example">
 <pre class="example">
+
 #include &quot;guicast.h&quot;
 
 MyWindow::MyWindow(MyPlugin *plugin)
@@ -10040,6 +10044,7 @@ the plugin header.  Instead, the following methods must be defined.
 </p>
 <div class="example">
 <pre class="example">
+
 BC_Hash *defaults;
 
 </pre></div>
@@ -10048,6 +10053,7 @@ BC_Hash *defaults;
 </p>
 <div class="example">
 <pre class="example">
+
 MainProgressBar *progress;
 </pre></div>
 
@@ -10102,6 +10108,7 @@ instantiate the progress object with a line like
 </p>
 <div class="example">
 <pre class="example">
+
 progress = start_progress(&quot;MyPlugin progress...&quot;, 
 	PluginClient::get_total_len());
 
@@ -10995,6 +11002,7 @@ Shift F   Set Format
 
 <div class="example">
 <pre class="example">
+
 Shift X    Cut keyframes
 Shift C    Copy keyframes
 Shift V    Paste keyframes
@@ -11010,6 +11018,7 @@ Alt v      Paste default keyframe
 
 <div class="example">
 <pre class="example">
+
 t          Add Audio Track
 u          Insert default Audio Transition
 Shift T    Add Video Track
@@ -11026,6 +11035,7 @@ Shift-Tab  Toggle every other track's arming status
 
 <div class="example">
 <pre class="example">
+
 1         Show titles
 2         Show transitions
 3         Fade keyframes
@@ -11047,6 +11057,7 @@ Shift-Tab  Toggle every other track's arming status
 
 <div class="example">
 <pre class="example">
+
 x         Cut
 c         Copy
 v         Paste
@@ -11092,6 +11103,7 @@ between the in/out points to be played, if in/out points are defined.
 
 <div class="example">
 <pre class="example">
+
 Space              Start and pause recording of the current batch
 l                  Toggle label at current position.
 
diff --git a/cinelerra-5.1/guicast/Makefile b/cinelerra-5.1/guicast/Makefile
index cf189de..fa4d3de 100644
--- a/cinelerra-5.1/guicast/Makefile
+++ b/cinelerra-5.1/guicast/Makefile
@@ -12,6 +12,11 @@ ifeq ($(OBJDIR), amd64)
 BOOTSTRAPFLAGS := -DBOOTSTRAP="\"objcopy -B i386 -I binary -O elf64-x86-64\""
 endif
 
+ifeq ($(OBJDIR), e2k)
+BOOTSTRAPFLAGS := -DBOOTSTRAP="\"objcopy -B e2k -I binary -O elf64-e2k\""
+endif
+
+
 $(shell mkdir -p $(OBJDIR) )
 
 OBJS = \
diff --git a/cinelerra-5.1/guicast/bcsignals.C b/cinelerra-5.1/guicast/bcsignals.C
index 858ba40..6586887 100644
--- a/cinelerra-5.1/guicast/bcsignals.C
+++ b/cinelerra-5.1/guicast/bcsignals.C
@@ -479,15 +479,16 @@ static void reg_dump(FILE *fp,sigregs_t *rp)
 static void reg_dump(FILE *fp,sigregs_t *rp) {}
 #endif
 
-#ifndef IP
-#error gotta have IP
-#endif
+//#ifndef IP
+//#error gotta have IP
+//#endif
 
 // HAVE_CTX
 #endif
 
 static void handle_dump(int n, siginfo_t * info, void *sc)
 {
+#if 0
 	uncatch_segv();  uncatch_intr();
 	signal(SIGSEGV, SIG_DFL);
 	signal(SIGINT, SIG_DFL);
@@ -595,5 +596,6 @@ static void handle_dump(int n, siginfo_t * info, void *sc)
         char *const argv[4] = { (char*) "/bin/sh", (char*) "-c", cmd, 0 };
         execvp(argv[0], &argv[0]);
 #endif
+#endif
 }
 
diff --git a/cinelerra-5.1/libzmpeg3/libzmpeg3.h b/cinelerra-5.1/libzmpeg3/libzmpeg3.h
index 430ebb4..9ebee66 100644
--- a/cinelerra-5.1/libzmpeg3/libzmpeg3.h
+++ b/cinelerra-5.1/libzmpeg3/libzmpeg3.h
@@ -5,7 +5,7 @@
 /* for quicktime build */
 #define MAXFRAMESAMPLES 65536
 #define ZDVB
-#define USE_FUTEX
+//#define USE_FUTEX
 
 #ifndef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS
diff --git a/cinelerra-5.1/mplexlo/Makefile b/cinelerra-5.1/mplexlo/Makefile
index c1a8aae..77b1c57 100644
--- a/cinelerra-5.1/mplexlo/Makefile
+++ b/cinelerra-5.1/mplexlo/Makefile
@@ -11,7 +11,7 @@ CFLAGS += -I$(LIBZMPEG3)
 CFLAGS += $(static_incs)
 
 LIBS := $(LIBZMPEG3)/$(OBJDIR)/libzmpeg3.a
-LIBS += $(libraries)
+LIBS += $(libraries) 
 
 OUTPUT := $(OBJDIR)/mplexlo
 
diff --git a/cinelerra-5.1/thirdparty/Makefile b/cinelerra-5.1/thirdparty/Makefile
index 27f13c6..051e979 100644
--- a/cinelerra-5.1/thirdparty/Makefile
+++ b/cinelerra-5.1/thirdparty/Makefile
@@ -130,19 +130,19 @@ esound.mak_vars+= CFLAGS=""
 esound.ldflags=" -lm -lstdc++"
 fftw.cfg_params= --disable-fortran --enable-shared=no
 ffmpeg.cfg_params= \
-	--enable-pthreads --enable-gpl --disable-ffplay \
+	--enable-pthreads --disable-crystalhd --enable-gpl --disable-ffplay \
 	$(call if_want,VAAPI,--enable-vaapi,--disable-vaapi) \
 	$(call if_want,VDPAU,--enable-vdpau,--disable-vdpau) \
 	$(call if_want,NV, --enable-nvenc --enable-nvdec --enable-ffnvcodec) \
 	$(call if_ena,twolame,--enable-libtwolame) \
 	$(call if_ena,openjpeg,--enable-libopenjpeg) \
-	$(call if_ena,lame,--enable-libmp3lame) \
+	$(call if_ena,lame,--disable-libmp3lame) \
 	$(call if_ena,libaom,--enable-libaom) \
 	$(call if_ena,dav1d,--enable-libdav1d) \
 	$(call if_ena,libwebp,--enable-libwebp) \
 	$(call if_ena,opus,--enable-libopus) \
 	$(call if_ena,libvorbis,--enable-libvorbis) \
-	$(call if_ena,libtheora,--enable-libtheora) \
+	$(call if_ena,libtheora,--disable-libtheora) \
 	$(call if_ena,libvpx,--enable-libvpx) \
 	$(call if_ena,x264,--enable-libx264) \
 	$(call if_ena,x265,--enable-libx265) \
@@ -183,9 +183,9 @@ ffmpeg.cfg_params= \
 
 #cmake_config=echo "exec cmake \$$$$@ $(1)" > ./configure; chmod +x ./configure;
 cmake_config=echo 'cmake "$$$$@" "$(1)"' > ./configure; chmod +x ./configure;
-a52dec.mak_params?= ; cd $(call bld_path,a52dec,include); ln -sf . a52dec
-a52dec.cfg_vars?= CFLAGS+=" -U__FreeBSD__ $(call inc_path,djbfft)" LIBS+=" $(call ld_path,djbfft)"
-a52dec.cfg_params?=--enable-djbfft
+a52dec.mak_params?= ; cd $(call bld_path,a52dec,include); ln -snf . a52dec
+a52dec.cfg_vars?= CFLAGS+=" -U__FreeBSD__ $(call inc_path,djbfft)" LIBS+=" $(call ld_path,djbfft)"; autoreconf -ifv && automake -caf;
+a52dec.cfg_params?=--disable-djbfft
 djbfft.cfg_vars?=echo "$(call bld_path,djbfft)" > conf-home; \
  (CFLAGS="$(CFLAGS)"; $(CFG_VARS)$(if $(CFG_VARS),; )echo "$(CC) $$$$CFLAGS") > conf-cc; \
  echo > ./configure; chmod +x ./configure;
@@ -197,7 +197,7 @@ flac.cflags?="$(call inc_path,libogg) $(call ld_path,libogg,src/.libs)"
 giflib.cfg_params=echo "exec true" > ./configure; chmod +x ./configure;
 lame.cfg_vars?= CFLAGS+=" -O"
 lame.cfg_params?=--enable-shared=no
-lame.mak_params?= ; cd $(call bld_path,lame,include); ln -sf . lame
+lame.mak_params?= ; cd $(call bld_path,lame,include); ln -snf . lame
 libaom.cfg_vars?= mkdir aom.build && cd aom.build && $(call cmake_config,..)
 libaom.cfg_params?= -DENABLE_SHARED=no -DCMAKE_INSTALL_LIBDIR=lib \
  -DCMAKE_INSTALL_PREFIX=$(call bld_path,libaom)/usr/local
@@ -208,48 +208,55 @@ libwebp.cfg_vars?= mkdir build && cd build && $(call cmake_config,..)
 libwebp.mak_params?= -C build all install DESTDIR=$(call bld_path,libwebp)
 mjpegtools.cflags?="$(call inc_path,libjpeg) $(call ld_path,libjpeg,build)"
 mjpegtools.cfg_vars?= ./autogen.sh; 
-mjpegtools.cfg_params?= --enable-shared=no --without-libquicktime --without-libdv \
+mjpegtools.cfg_params?= --enable-shared=no --disable-simd-accel --without-libquicktime --without-libdv \
  --without-libpng --without-dga --without-gtk --without-libsdl --without-sdlgfx
 mjpegtools.mak_params?= all
-ladspa.cfg_vars?= CFLAGS+=' -Dinline="" '
+ladspa.cfg_vars?= CFLAGS+=' -Dinline="" '; autoreconf -ifv && automake -caf;
 ladspa.mak_params?=; $(MAKE) -C ladspa* install DESTDIR=$(call bld_path,ladspa)
-libavc1394.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394)
+libavc1394.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394);autoreconf -ifv -I m4 && automake -caf;
 libavc1394.cflags?="$(call inc_path,libraw1394)"
 libavc1394.ldflags?="$(call ld_path,libraw1394,src/.libs)"
 libavc1394.cfg_params?= --enable-shared=no
-libavc1394.mak_params?=; cd $(call bld_path,libavc1394); ln -sf ../librom1394/rom1394.h libavc1394/.
+libavc1394.mak_params?=; cd $(call bld_path,libavc1394); ln -snf ../librom1394/rom1394.h libavc1394/.
+libdv.cfg_vars?= autoreconf -ifv -I m4 && automake -caf;
 libdv.cfg_params?= --disable-gtk --enable-shared=no
-libiec61883.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394)
+libiec61883.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394);autoreconf -ifv -I m4 && automake -caf;
 libiec61883.cflags?="$(call inc_path,libraw1394)"
 libiec61883.ldflags?="$(call ld_path,libraw1394,src/.libs)"
 libiec61883.cfg_params?= --enable-shared=no
-libiec61883.mak_params?=; cd $(call bld_path,libiec61883,src); ln -sf . libiec61883
+libiec61883.mak_params?=; cd $(call bld_path,libiec61883,src); ln -snf . libiec61883
 libjpeg.cfg_vars?= mkdir build && cd build && $(call cmake_config,..)
 libjpeg.cfg_params?= -DENABLE_SHARED=no -DCMAKE_INSTALL_LIBDIR=lib 
 libjpeg.mak_params?= -C build all install DESTDIR=$(call bld_path,libjpeg)
+libogg.cfg_vars?= autoreconf -ifv -I m4 && automake -caf;
 libogg.cfg_params?= --enable-shared=no
-libraw1394.cfg_params?=  --enable-shared=no; ln -sf src libraw1394
-libtheora.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libogg):$(call bld_path,libvorbis)
+libraw1394.cfg_vars?= autoreconf -ifv -I m4 && automake -caf;
+libraw1394.cfg_params?=  --enable-shared=no; ln -snf src libraw1394
+libtheora.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libogg):$(call bld_path,libvorbis); autoreconf -ifv -I m4 && automake -caf;
 libtheora.cflags?="$(call inc_path,libogg) $(call inc_path,libvorbis)"
 libtheora.ldflags?="$(call ld_path,libvorbis,lib/.libs) $(call ld_path,libogg,src/.libs)"
 libtheora.cfg_params?= --disable-examples --disable-spec --enable-shared=no
+libuuid.cfg_vars?= autoreconf -ifv -I m4 && automake -caf;
 libuuid.cfg_params?=--enable-shared=no
 libvorbis.cfg_params?= --disable-oggtest --enable-shared=no
+nasm.cfg_params?= --disable-doc
 openjpeg.cfg_params?= -DBUILD_SHARED_LIBS:BOOL=OFF
 openjpeg.cfg_vars?=$(call cmake_config,.)
-openjpeg.mak_params?= ; cd $(call bld_path,openjpeg,src/lib/openjp2); ln -sf . openjpeg-2.1
+openjpeg.mak_params?= ; cd $(call bld_path,openjpeg,src/lib/openjp2); ln -snf . openjpeg-2.1
 opencv.cfg_vars?=$(call cmake_config,.)
 opencv.cfg_params?= -DBUILD_SHARED_LIBS:BOOL=OFF
 openexr.cfg_vars?=true; \#
 openexr.mak_vars?=true; \#
+opus.cfg_params?= --disable-doc  --disable-extra-programs 
 speech_tools.mak_params?=-j1
-tiff.cfg_vars+=LIBS+=" -lpthread"
-tiff.cfg_params+= --enable-shared=no --disable-zstd $(call if_pkg,libwebp,\
+tiff.cfg_vars+=LIBS+=" -lpthread"; autoreconf -ifv -I m4 && automake -caf;
+tiff.cfg_params+= --enable-shared=no --disable-webp --disable-zstd $(call if_pkg,libwebp,\
  --with-webp-include-dir=$(call pkg_incl,libwebp)\
  --with-webp-lib-dir=$(call pkg_libs,libwebp))\
  $(call if_npkg,libwebp,--disable-webp)
 twolame.cfg_params?=--enable-shared=no
-x264.cfg_params?= --enable-static --enable-pic
+x264.cfg_vars?= cp /usr/share/automake-1.16/config.* .;
+x264.cfg_params?= --enable-static --enable-pic --disable-asm
 x265.cfg_vars?=$(call cmake_config,source)
 x265.cfg_params?= -DENABLE_SHARED=no
 libvpx.cfg_params?= --enable-pic --disable-avx512
@@ -316,6 +323,7 @@ else
 rules=$(eval $(1))
 endif
 
+$(call rules,$(call std-build,nasm))
 $(call rules,$(call std-build,a52dec,djbfft))
 $(call rules,$(call std-build,djbfft))
 $(call rules,$(call std-build,audiofile))
@@ -329,9 +337,9 @@ $(call rules,$(call std-build,flac,libogg))
 $(call rules,$(call std-build,giflib))
 $(call rules,$(call std-build,ilmBase, openexr))
 $(call rules,$(call std-build,ladspa))
-$(call rules,$(call std-build,lame))
+$(call rules,$(call std-build,lame,nasm))
 $(call rules,$(call std-build,libaom))
-$(call rules,$(call std-build,dav1d))
+$(call rules,$(call std-build,dav1d,nasm))
 $(call rules,$(call std-build,libwebp))
 $(call rules,$(call std-build,libavc1394,libraw1394))
 $(call rules,$(call std-build,libdv))
@@ -352,8 +360,8 @@ $(call rules,$(call std-build,opus))
 $(call rules,$(call std-build,speech_tools))
 $(call rules,$(call std-build,tiff, libwebp))
 $(call rules,$(call std-build,twolame))
-$(call rules,$(call std-build,x264))
-$(call rules,$(call std-build,x265))
+$(call rules,$(call std-build,x264,nasm))
+$(call rules,$(call std-build,x265,nasm))
 $(call rules,$(call std-build,libvpx))
 $(call rules,$(call std-build,lv2))
 $(call rules,$(call std-build,serd))
