diff --git a/cinelerra-5.1/cinelerra/mwindow.C b/cinelerra-5.1/cinelerra/mwindow.C
index 04549fde..ded1603e 100644
--- a/cinelerra-5.1/cinelerra/mwindow.C
+++ b/cinelerra-5.1/cinelerra/mwindow.C
@@ -2211,8 +2211,9 @@ if(debug) printf("MWindow::load_filenames %d\n", __LINE__);
 			if( strcmp(cin_version, CINELERRA_VERSION) &&
 			    strcmp(cin_version, "Unify") &&
 			    strcmp(cin_version, "5.1") ) {
+				if (preferences->warn_version)
 				eprintf(_("Warning: XML from cinelerra version %s\n"
-					"Session data may be incompatible."), cin_version);
+                                        "Session data may be incompatible."), cin_version);
 			}
 			if( new_edl->load_xml(&xml_file, LOAD_ALL) ) {
 				eprintf(_("Error: unable to load:\n  %s"), filename);
