diff --git a/cinelerra-5.1/cinelerra/menuattachtransition.C b/cinelerra-5.1/cinelerra/menuattachtransition.C
index efd0a823..ae2c959d 100644
--- a/cinelerra-5.1/cinelerra/menuattachtransition.C
+++ b/cinelerra-5.1/cinelerra/menuattachtransition.C
@@ -82,8 +82,11 @@ void TransitionDialogThread::start()
 		mwindow->search_plugindb(data_type == TRACK_AUDIO,
 			data_type == TRACK_VIDEO, 0, 1, 0, plugindb);
 		for(int i = 0; i < plugindb.total; i++) {
-			const char *title = plugindb.values[i]->title;
-			if( !strcmp(transition_title, title) ) number = i;
+			char *title = _(plugindb.values[i]->title);
+			//printf("Title: %s\n", title);
+			mwindow->fix_plugin_title(title);
+			//printf("Title_fixed: %s\n", title);
+			if( !strcmp(transition_title, title)) number = i;
 			transition_names.append(new BC_ListBoxItem(_(title)));
 		}
 	}
