--- configure.ac.orig	2019-12-01 10:13:05.312067175 +0300
+++ configure.ac	2019-12-01 10:17:56.697491685 +0300
@@ -914,7 +914,7 @@
 FFMPEG_EXTRA_LDFLAGS=""
 if test "x$WANT_VAAPI" != "xno" -a "x$HAVE_VAAPI" = "xyes"; then
   FFMPEG_EXTRA_LDFLAGS+=' -lva'
-  EXTRA_LIBS+=' -lva'
+  EXTRA_LIBS+=' -lva -L/usr/X11R7/lib'
   WANT_VAAPI="yes"
   if test "x$HAVE_vaapi_x11" = "xyes"; then
     FFMPEG_EXTRA_LDFLAGS+=' -lva-x11'
