diff --git a/cinelerra-5.1/cinelerra/mainerror.h b/cinelerra-5.1/cinelerra/mainerror.h
index 766f1e66..69700807 100644
--- a/cinelerra-5.1/cinelerra/mainerror.h
+++ b/cinelerra-5.1/cinelerra/mainerror.h
@@ -89,7 +89,7 @@ private:
 
 // format text to error dialog listbox
 
-static inline void eprint1(const char *func, const char *fmt, ...)
+static  void eprint1(const char *func, const char *fmt, ...)
 {
 	char err_msg[1024], *cp = err_msg, *ep = cp + sizeof(err_msg)-1;
 	va_list va;
