diff --git a/cinelerra-5.1/thirdparty/Makefile b/cinelerra-5.1/thirdparty/Makefile
index 27f13c65..a3d3b0c0 100644
--- a/cinelerra-5.1/thirdparty/Makefile
+++ b/cinelerra-5.1/thirdparty/Makefile
@@ -131,6 +131,7 @@ esound.ldflags=" -lm -lstdc++"
 fftw.cfg_params= --disable-fortran --enable-shared=no
 ffmpeg.cfg_params= \
 	--enable-pthreads --enable-gpl --disable-ffplay \
+	--disable-debug --disable-programs --disable-doc \
 	$(call if_want,VAAPI,--enable-vaapi,--disable-vaapi) \
 	$(call if_want,VDPAU,--enable-vdpau,--disable-vdpau) \
 	$(call if_want,NV, --enable-nvenc --enable-nvdec --enable-ffnvcodec) \
@@ -183,8 +184,8 @@ ffmpeg.cfg_params= \
 
 #cmake_config=echo "exec cmake \$$$$@ $(1)" > ./configure; chmod +x ./configure;
 cmake_config=echo 'cmake "$$$$@" "$(1)"' > ./configure; chmod +x ./configure;
-a52dec.mak_params?= ; cd $(call bld_path,a52dec,include); ln -sf . a52dec
-a52dec.cfg_vars?= CFLAGS+=" -U__FreeBSD__ $(call inc_path,djbfft)" LIBS+=" $(call ld_path,djbfft)"
+a52dec.mak_params?= ; cd $(call bld_path,a52dec,include); ln -snf . a52dec
+a52dec.cfg_vars?= CFLAGS+=" -U__FreeBSD__ $(call inc_path,djbfft)" LIBS+=" $(call ld_path,djbfft)"; autoreconf -ifv && automake -caf;
 a52dec.cfg_params?=--enable-djbfft
 djbfft.cfg_vars?=echo "$(call bld_path,djbfft)" > conf-home; \
  (CFLAGS="$(CFLAGS)"; $(CFG_VARS)$(if $(CFG_VARS),; )echo "$(CC) $$$$CFLAGS") > conf-cc; \
@@ -195,11 +196,11 @@ audiofile.mak_params?=LIBS="-lm -lstdc++"
 flac.cfg_params?= --enable-shared=no
 flac.cflags?="$(call inc_path,libogg) $(call ld_path,libogg,src/.libs)"
 giflib.cfg_params=echo "exec true" > ./configure; chmod +x ./configure;
-lame.cfg_vars?= CFLAGS+=" -O"
-lame.cfg_params?=--enable-shared=no
-lame.mak_params?= ; cd $(call bld_path,lame,include); ln -sf . lame
+lame.cfg_vars?= CFLAGS+=" -O3 -DUSE_FAST_LOG=1 -ffast-math"
+lame.cfg_params?=--enable-shared=no --enable-nasm --enable-expopt=norm
+lame.mak_params?= ; cd $(call bld_path,lame,include); ln -snf . lame
 libaom.cfg_vars?= mkdir aom.build && cd aom.build && $(call cmake_config,..)
-libaom.cfg_params?= -DENABLE_SHARED=no -DCMAKE_INSTALL_LIBDIR=lib \
+libaom.cfg_params?= -DENABLE_SHARED=no -DENABLE_TESTS=OFF -DENABLE_EXAMPLES=OFF -DENABLE_TOOLS=OFF -DCMAKE_INSTALL_LIBDIR=lib \
  -DCMAKE_INSTALL_PREFIX=$(call bld_path,libaom)/usr/local
 libaom.mak_params?= ; $(MAKE) -C libaom*/aom.build install
 dav1d.cfg_vars?=echo "echo dav1d custom make" >> configure; chmod +x ./configure;
@@ -211,37 +212,41 @@ mjpegtools.cfg_vars?= ./autogen.sh;
 mjpegtools.cfg_params?= --enable-shared=no --without-libquicktime --without-libdv \
  --without-libpng --without-dga --without-gtk --without-libsdl --without-sdlgfx
 mjpegtools.mak_params?= all
-ladspa.cfg_vars?= CFLAGS+=' -Dinline="" '
+ladspa.cfg_vars?= CFLAGS+=' -Dinline="" '; autoreconf -ifv && automake -caf;
 ladspa.mak_params?=; $(MAKE) -C ladspa* install DESTDIR=$(call bld_path,ladspa)
 libavc1394.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394)
 libavc1394.cflags?="$(call inc_path,libraw1394)"
 libavc1394.ldflags?="$(call ld_path,libraw1394,src/.libs)"
 libavc1394.cfg_params?= --enable-shared=no
-libavc1394.mak_params?=; cd $(call bld_path,libavc1394); ln -sf ../librom1394/rom1394.h libavc1394/.
+libavc1394.mak_params?=; cd $(call bld_path,libavc1394); ln -snf ../librom1394/rom1394.h libavc1394/.
+libdv.cfg_vars?=; autoreconf -ifv -I m4 && automake -caf;
 libdv.cfg_params?= --disable-gtk --enable-shared=no
 libiec61883.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libraw1394)
 libiec61883.cflags?="$(call inc_path,libraw1394)"
 libiec61883.ldflags?="$(call ld_path,libraw1394,src/.libs)"
 libiec61883.cfg_params?= --enable-shared=no
-libiec61883.mak_params?=; cd $(call bld_path,libiec61883,src); ln -sf . libiec61883
+libiec61883.mak_params?=; cd $(call bld_path,libiec61883,src); ln -snf . libiec61883
 libjpeg.cfg_vars?= mkdir build && cd build && $(call cmake_config,..)
 libjpeg.cfg_params?= -DENABLE_SHARED=no -DCMAKE_INSTALL_LIBDIR=lib 
 libjpeg.mak_params?= -C build all install DESTDIR=$(call bld_path,libjpeg)
 libogg.cfg_params?= --enable-shared=no
-libraw1394.cfg_params?=  --enable-shared=no; ln -sf src libraw1394
-libtheora.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libogg):$(call bld_path,libvorbis)
+libraw1394.cfg_params?=  --enable-shared=no; ln -snf src libraw1394
+libtheora.cfg_vars?=PKG_CONFIG_PATH=$(call bld_path,libogg):$(call bld_path,libvorbis); autoreconf -ifv -I m4 && automake -caf;
 libtheora.cflags?="$(call inc_path,libogg) $(call inc_path,libvorbis)"
 libtheora.ldflags?="$(call ld_path,libvorbis,lib/.libs) $(call ld_path,libogg,src/.libs)"
 libtheora.cfg_params?= --disable-examples --disable-spec --enable-shared=no
+libuuid.cfg_vars?= autoreconf -ifv -I m4 && automake -caf;
 libuuid.cfg_params?=--enable-shared=no
 libvorbis.cfg_params?= --disable-oggtest --enable-shared=no
+nasm.cfg_params?= --disable-doc
 openjpeg.cfg_params?= -DBUILD_SHARED_LIBS:BOOL=OFF
 openjpeg.cfg_vars?=$(call cmake_config,.)
-openjpeg.mak_params?= ; cd $(call bld_path,openjpeg,src/lib/openjp2); ln -sf . openjpeg-2.1
+openjpeg.mak_params?= ; cd $(call bld_path,openjpeg,src/lib/openjp2); ln -snf . openjpeg-2.1
 opencv.cfg_vars?=$(call cmake_config,.)
 opencv.cfg_params?= -DBUILD_SHARED_LIBS:BOOL=OFF
 openexr.cfg_vars?=true; \#
 openexr.mak_vars?=true; \#
+opus.cfg_params?= --disable-doc  --disable-extra-programs 
 speech_tools.mak_params?=-j1
 tiff.cfg_vars+=LIBS+=" -lpthread"
 tiff.cfg_params+= --enable-shared=no --disable-zstd $(call if_pkg,libwebp,\
@@ -250,9 +255,11 @@ tiff.cfg_params+= --enable-shared=no --disable-zstd $(call if_pkg,libwebp,\
  $(call if_npkg,libwebp,--disable-webp)
 twolame.cfg_params?=--enable-shared=no
 x264.cfg_params?= --enable-static --enable-pic
-x265.cfg_vars?=$(call cmake_config,source)
-x265.cfg_params?= -DENABLE_SHARED=no
-libvpx.cfg_params?= --enable-pic --disable-avx512
+x265.cfg_vars?=chmod +x ./configure; chmod +x ./multilib.sh;
+#x265.cfg_vars?=$(call cmake_config,source)
+#x265.cfg_params?= -DENABLE_SHARED=no
+libvpx.cfg_params?= --enable-pic --disable-avx512 --disable-unit-tests --disable-examples --disable-tools
+
 
 DS:=$$$$$$$$
 pkg_cfg=$(call bld_path,$(1),usr/local/lib/pkgconfig):
@@ -316,6 +323,7 @@ else
 rules=$(eval $(1))
 endif
 
+$(call rules,$(call std-build,nasm))
 $(call rules,$(call std-build,a52dec,djbfft))
 $(call rules,$(call std-build,djbfft))
 $(call rules,$(call std-build,audiofile))
@@ -329,9 +337,9 @@ $(call rules,$(call std-build,flac,libogg))
 $(call rules,$(call std-build,giflib))
 $(call rules,$(call std-build,ilmBase, openexr))
 $(call rules,$(call std-build,ladspa))
-$(call rules,$(call std-build,lame))
+$(call rules,$(call std-build,lame,nasm))
 $(call rules,$(call std-build,libaom))
-$(call rules,$(call std-build,dav1d))
+$(call rules,$(call std-build,dav1d,nasm))
 $(call rules,$(call std-build,libwebp))
 $(call rules,$(call std-build,libavc1394,libraw1394))
 $(call rules,$(call std-build,libdv))
@@ -352,8 +360,8 @@ $(call rules,$(call std-build,opus))
 $(call rules,$(call std-build,speech_tools))
 $(call rules,$(call std-build,tiff, libwebp))
 $(call rules,$(call std-build,twolame))
-$(call rules,$(call std-build,x264))
-$(call rules,$(call std-build,x265))
+$(call rules,$(call std-build,x264,nasm))
+$(call rules,$(call std-build,x265,nasm))
 $(call rules,$(call std-build,libvpx))
 $(call rules,$(call std-build,lv2))
 $(call rules,$(call std-build,serd))
