diff --git a/cinelerra-5.1/opencv_build b/cinelerra-5.1/opencv_build
index 20d6ea1..1e4c777 100644
--- a/cinelerra-5.1/opencv_build
+++ b/cinelerra-5.1/opencv_build
@@ -76,6 +76,7 @@ $(opencv)/build: $(opencv).src
   -DCMAKE_BUILD_TYPE=RELEASE \
   -DWITH_IPP=OFF \
   -DWITH_LAPACK=OFF \
+  -DWITH_GPHOTO2=OFF \
   -DBUILD_SHARED_LIBS=OFF \
   -DINSTALL_C_EXAMPLES=OFF \
   -DINSTALL_PYTHON_EXAMPLES=OFF \
@@ -97,6 +98,7 @@ $(opencv)/build: $(opencv).src
   -DCMAKE_BUILD_TYPE=RELEASE \
   -DWITH_IPP=OFF \
   -DWITH_LAPACK=OFF \
+  -DWITH_GPHOTO2=OFF \
   -DBUILD_SHARED_LIBS=ON \
   -DINSTALL_C_EXAMPLES=ON \
   -DINSTALL_PYTHON_EXAMPLES=ON \
