--- thirdparty/Makefile.orig	2023-12-22 11:55:25.898670705 +0300
+++ thirdparty/Makefile	2023-12-22 11:57:20.746672566 +0300
@@ -130,6 +130,7 @@
 esound.ldflags=" -lm -lstdc++"
 fftw.cfg_params= --disable-fortran --enable-shared=no
 ffmpeg.cfg_vars?= PKG_CONFIG_PATH="$(call bld_path,libwebp)/usr/local/lib/pkgconfig"
+ffmpeg.cfg_vars?+= CFLAGS+="-I/usr/include/svt-av1"
 ffmpeg.cfg_params= \
 	--enable-pthreads --disable-avdevice --enable-gpl --disable-ffplay \
 	--disable-doc \
