diff --git a/cinelerra-5.1/cinelerra/ffmpeg.C b/cinelerra-5.1/cinelerra/ffmpeg.C
index 36fc196d..fae5a104 100644
--- a/cinelerra-5.1/cinelerra/ffmpeg.C
+++ b/cinelerra-5.1/cinelerra/ffmpeg.C
@@ -1208,18 +1208,20 @@ AVHWDeviceType FFVideoStream::encode_hw_activate(const char *hw_dev, const char
 	AVHWDeviceType type = AV_HWDEVICE_TYPE_NONE;
 	if( strcmp(_("none"), hw_dev) ) {
 		type = av_hwdevice_find_type_by_name(hw_dev);
-		if( type != AV_HWDEVICE_TYPE_VAAPI ) {
-			fprintf(stderr, "currently, only vaapi hw encode is supported\n");
+		if( !((type == AV_HWDEVICE_TYPE_VAAPI) || (type == AV_HWDEVICE_TYPE_VULKAN ))) {
+			fprintf(stderr, "currently, only vaapi/vulkan hw encode is supported\n");
 			type = AV_HWDEVICE_TYPE_NONE;
 		}
 	}
+
 	if( type != AV_HWDEVICE_TYPE_NONE ) {
 		int ret = 0;
 		if (drm_node_enc) {
 		ret = av_hwdevice_ctx_create(&hw_device_ctx, AV_HWDEVICE_TYPE_VAAPI, drm_node_enc, 0, 0);
 		} else {
-		ret = av_hwdevice_ctx_create(&hw_device_ctx, AV_HWDEVICE_TYPE_VAAPI, 0, 0, 0);
+		ret = av_hwdevice_ctx_create(&hw_device_ctx, type, 0, 0, 0);
 		}
+		
 		if( ret < 0 ) {
 			ff_err(ret, "Failed to create a HW device.\n");
 			type = AV_HWDEVICE_TYPE_NONE;
@@ -1229,21 +1231,27 @@ AVHWDeviceType FFVideoStream::encode_hw_activate(const char *hw_dev, const char
 		hw_frames_ref = av_hwframe_ctx_alloc(hw_device_ctx);
 		if( !hw_frames_ref ) {
 			fprintf(stderr, "Failed to create HW frame context.\n");
-			type = AV_HWDEVICE_TYPE_NONE;
+			type = type;
 		}
 	}
 	if( type != AV_HWDEVICE_TYPE_NONE ) {
 		AVHWFramesContext *frames_ctx = (AVHWFramesContext *)(hw_frames_ref->data);
 		frames_ctx->format = AV_PIX_FMT_VAAPI;
+
+		if (type == AV_HWDEVICE_TYPE_VULKAN)
+		frames_ctx->format = AV_PIX_FMT_VULKAN;
+
 		frames_ctx->sw_format = AV_PIX_FMT_NV12;
 		if (strcmp(hw_sformat, "vaapi")) frames_ctx->sw_format = av_get_pix_fmt(hw_sformat);
+		if (strcmp(hw_sformat, "vulkan")) frames_ctx->sw_format = av_get_pix_fmt(hw_sformat);
+		
 		frames_ctx->width = width;
 		frames_ctx->height = height;
 		frames_ctx->initial_pool_size = 0; // 200;
 		int ret = av_hwframe_ctx_init(hw_frames_ref);
 		if( ret >= 0 ) {
 			avctx->hw_frames_ctx = av_buffer_ref(hw_frames_ref);
-			if( !avctx->hw_frames_ctx ) ret = AVERROR(ENOMEM);
+						if( !avctx->hw_frames_ctx ) ret = AVERROR(ENOMEM);
 		}
 		if( ret < 0 ) {
 			ff_err(ret, "Failed to initialize HW frame context.\n");
@@ -1260,6 +1268,7 @@ int FFVideoStream::encode_hw_write(FFrame *picture)
 	AVFrame *hw_frm = 0;
 	switch( avctx->pix_fmt ) {
 	case AV_PIX_FMT_VAAPI:
+	case AV_PIX_FMT_VULKAN:
 		hw_frm = av_frame_alloc();
 		if( !hw_frm ) { ret = AVERROR(ENOMEM);  break; }
 		ret = av_hwframe_get_buffer(avctx->hw_frames_ctx, hw_frm, 0);
@@ -1411,6 +1420,9 @@ int FFVideoStream::init_frame(AVFrame *picture)
 	case AV_PIX_FMT_VAAPI:
 		picture->format = AV_PIX_FMT_NV12;
 		break;
+	case AV_PIX_FMT_VULKAN:
+		picture->format = avctx->sw_pix_fmt;
+		break;
 	default:
 		picture->format = avctx->pix_fmt;
 		break;
@@ -1499,11 +1511,13 @@ int FFVideoStream::encode(VFrame *vframe)
 	ffmpeg->flow_ctl();
 	FFrame *picture = new FFrame(this);
 	int ret = picture->initted();
+//	printf("FFVideoStream::encode: after picture->initted() ret = %i \n", ret);
 	if( ret >= 0 ) {
 		AVFrame *frame = *picture;
 		frame->pts = curr_pos;
 		ret = convert_pixfmt(vframe, frame);
 	}
+
 	if( ret >= 0 && avctx->hw_frames_ctx )
 		encode_hw_write(picture);
 	if( ret >= 0 ) {
@@ -1536,7 +1550,7 @@ int FFVideoStream::encode_frame(AVFrame *frame)
 		frame->flags |= AV_FRAME_FLAG_INTERLACED;
 #endif
 	}
-	if( frame && frame->format == AV_PIX_FMT_VAAPI ) { // ugly
+	if( frame && (frame->format == AV_PIX_FMT_VAAPI || frame->format == AV_PIX_FMT_VULKAN))  { // ugly
 		int ret = avcodec_send_frame(avctx, frame);
 		for( int retry=MAX_RETRY; !ret && --retry>=0; ) {
 			FFPacket pkt;  av_init_packet(pkt);
@@ -1659,7 +1673,13 @@ int FFVideoConvert::convert_picture_vframe(VFrame *frame, AVFrame *ip, AVFrame *
 
 	AVPixelFormat pix_fmt = (AVPixelFormat)ip->format;
 	FFVideoStream *vid =(FFVideoStream *)this;
+	if (pix_fmt == AV_PIX_FMT_VULKAN)
+//	printf("Vulkan1 \n");
+	
+//	printf("In FFVideoConvert::convert_picture_vframe! \n");
+	
 	if( pix_fmt == vid->hw_pixfmt ) {
+//		printf("FFVideoConvert::convert_picture_vframe called with hw_pixfmt! \n");
 		int ret = 0;
 		if( !sw_frame && !(sw_frame=av_frame_alloc()) )
 			ret = AVERROR(ENOMEM);
@@ -1850,6 +1870,7 @@ int FFVideoConvert::convert_vframe_picture(VFrame *frame, AVFrame *op, AVFrame *
 
 int FFVideoConvert::convert_pixfmt(VFrame *frame, AVFrame *op)
 {
+//	printf("in FVideoConvert::convert_pixfmt! \n");
 	// try direct transfer
 	if( !convert_vframe_picture(frame, op) ) return 1;
 	// use indirect transfer
@@ -3091,12 +3112,20 @@ int FFMPEG::open_encoder(const char *type, const char *spec)
 			case BC_COLORS_BT2020_CL: ctx->colorspace = AVCOL_SPC_BT2020_CL; break;
 			}
 			AVPixelFormat pix_fmt = av_get_pix_fmt(asset->ff_pixel_format);
-			if( opt_hw_dev != 0 ) {
-				AVHWDeviceType hw_type = vid->encode_hw_activate(opt_hw_dev, asset->ff_pixel_format);
+			AVHWDeviceType hw_type;
+			if( opt_hw_dev != 0 || ff_hw_dev() != 0 ) {
+			if(opt_hw_dev)
+				hw_type = vid->encode_hw_activate(opt_hw_dev, asset->ff_pixel_format);
+			if(ff_hw_dev())
+				hw_type = vid->encode_hw_activate(ff_hw_dev(), asset->ff_pixel_format);
 				switch( hw_type ) {
 				case AV_HWDEVICE_TYPE_VAAPI:
 					pix_fmt = AV_PIX_FMT_VAAPI;
 					break;
+				case AV_HWDEVICE_TYPE_VULKAN:
+					pix_fmt = AV_PIX_FMT_VULKAN;
+					//printf("Vulkan pixfmt! \n");
+					break;
 				case AV_HWDEVICE_TYPE_NONE:
 				default: break;
 				}
