From db6c336b8df73b64a744e066c841321ab9b1947a Mon Sep 17 00:00:00 2001
From: Andrew Randrianasulu <randrianasulu@gmail.com>
Date: Fri, 20 Mar 2026 10:37:28 +0300
Subject: [PATCH 9/9] Possible fix for parallel x265 make

---
 cinelerra-5.1/thirdparty/src/x265_4.0.patch3 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cinelerra-5.1/thirdparty/src/x265_4.0.patch3 b/cinelerra-5.1/thirdparty/src/x265_4.0.patch3
index ecca4bcd..3e0f6020 100644
--- a/cinelerra-5.1/thirdparty/src/x265_4.0.patch3
+++ b/cinelerra-5.1/thirdparty/src/x265_4.0.patch3
@@ -14,7 +14,7 @@
 +  # 32-bit stuff here
 +cmake ../source -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DMAIN12=ON
 +fi
-+make
++make -j
 +
 +cd ../10bit
 +if [ $(uname -m) == 'x86_64' ]; then
@@ -24,13 +24,13 @@
 +  # 32-bit stuff here
 +cmake ../source -DHIGH_BIT_DEPTH=ON -DENABLE_ASSEMBLY=OFF -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF
 +fi
-+make
++make -j
 +
 +cd ../8bit
 +ln -sf ../10bit/libx265.a libx265_main10.a
 +ln -sf ../12bit/libx265.a libx265_main12.a
 +cmake ../source -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DENABLE_SHARED=OFF -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DENABLE_CLI=OFF
-+make
++make -j
 +
 +# rename the 8bit library, then combine all three into libx265.a
 +mv libx265.a libx265_main.a
-- 
2.46.4

