From 49e3834a0108a845ece885a1bccca0883fb16010 Mon Sep 17 00:00:00 2001
From: Andrew Randrianasulu <randrianasulu@gmail.com>
Date: Fri, 19 Apr 2024 06:05:34 +0300
Subject: [PATCH 2/3] Fix ffmpeg audio filter harder?

---
 cinelerra-5.1/cinelerra/pluginfclient.C | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/cinelerra-5.1/cinelerra/pluginfclient.C b/cinelerra-5.1/cinelerra/pluginfclient.C
index 9cf6407f..0ffcf905 100644
--- a/cinelerra-5.1/cinelerra/pluginfclient.C
+++ b/cinelerra-5.1/cinelerra/pluginfclient.C
@@ -858,10 +858,11 @@ int PluginFAClient::activate()
 	if( ret >= 0 && ffilt->filter->inputs ) {
 		char args[BCTEXTLEN];
 		snprintf(args, sizeof(args),
-			"time_base=%d/%d:sample_rate=%d:sample_fmt=%s:channel_layout=%s",
-			1, sample_rate, sample_rate, av_get_sample_fmt_name(sample_fmt), chLayoutDescription);
+			"time_base=%d/%d:sample_rate=%d:sample_fmt=%s:channel_layout=%s:channels=%i",
+			1, sample_rate, sample_rate, av_get_sample_fmt_name(sample_fmt), chLayoutDescription, channels);
 		ret = avfilter_graph_create_filter(&fsrc, avfilter_get_by_name("abuffer"),
 			"in", args, NULL, graph);
+		if(ret <0) printf("abuffer failed!\n");
 	}
 	if( ret >= 0 )
 		ret = avfilter_graph_create_filter(&fsink, avfilter_get_by_name("abuffersink"),
@@ -869,9 +870,9 @@ int PluginFAClient::activate()
 	if( ret >= 0 )
 		ret = av_opt_set_bin(fsink, "sample_fmts",
 			(uint8_t*)&sample_fmt, sizeof(sample_fmt), AV_OPT_SEARCH_CHILDREN);
-	if( ret >= 0 )
-		ret = av_opt_set_bin(fsink, "channel_layout",
-			(uint8_t*)&layout, sizeof(layout), AV_OPT_SEARCH_CHILDREN);
+/*	if( ret >= 0 )
+		ret = av_opt_set_bin(fsink, "ch_layouts",
+			(uint8_t*)&chLayoutDescription, sizeof(chLayoutDescription), AV_OPT_SEARCH_CHILDREN); */
 	if( ret >= 0 )
 		ret = av_opt_set_bin(fsink, "sample_rates",
 			(uint8_t*)&sample_rate, sizeof(sample_rate), AV_OPT_SEARCH_CHILDREN);
-- 
2.35.8

