--- /home/guest/botva/src/cinelerra-git/cin-5/new-git/cinelerra/cinelerra-5.1/guicast/bootstrap.c	2022-09-10 19:25:45.386945418 +0300
+++ /dev/shm/bootstrap.c	2025-12-19 19:33:55.342785491 +0300
@@ -53,8 +53,16 @@
 	strcpy(buffer + *buffer_size, string);
 
 	*buffer_size += strlen(string) + 1;
-
-	*(int*)(buffer + *buffer_size) = data_offset;
+	
+	int tmp = 0;
+	char *ptr = buffer + *buffer_size;
+	memcpy(&tmp, &ptr, sizeof(int));
+	tmp  = data_offset;
+	ptr[3] = tmp >> 24;
+	ptr[2] = tmp >> 16;
+	ptr[1] = tmp >> 8;
+	ptr[0] = tmp;
+	//*(int*)(buffer + *buffer_size) = data_offset;
 	*buffer_size += sizeof(int);
 }
 
