From 76632cfe8671fbea24e5950d604dfc1fe886ac14 Mon Sep 17 00:00:00 2001
From: Andrew Randrianasulu <randrianasulu@gmail.com>
Date: Tue, 16 Jan 2024 10:01:29 +0300
Subject: [PATCH 4/4] Make 1440 formats also 16:9

---
 cinelerra-5.1/cinelerra/bdcreate.C | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cinelerra-5.1/cinelerra/bdcreate.C b/cinelerra-5.1/cinelerra/bdcreate.C
index dac7e442..13a062ad 100644
--- a/cinelerra-5.1/cinelerra/bdcreate.C
+++ b/cinelerra-5.1/cinelerra/bdcreate.C
@@ -80,10 +80,10 @@ static struct bd_format {
 	{ "1920x1080 25i",	1920,1080, 25.,    1, ILACE_MODE_TOP_FIRST },
 	{ "1920x1080 25p*",     1920,1080, 25.,    1, ILACE_MODE_FAKE_INTERLACE },
 	{ "1920x1080 23.976p",	1920,1080, 23.976, 1, ILACE_MODE_NOTINTERLACED },
-	{ "1440x1080 29.97i",	1440,1080, 29.97, -1, ILACE_MODE_TOP_FIRST },
-	{ "1440x1080 25i",	1440,1080, 25.,	  -1, ILACE_MODE_TOP_FIRST },
-	{ "1440x1080 24p",	1440,1080, 24.,	  -1, ILACE_MODE_NOTINTERLACED },
-	{ "1440x1080 23.976p",	1440,1080, 23.976,-1, ILACE_MODE_NOTINTERLACED },
+	{ "1440x1080 29.97i",	1440,1080, 29.97,  1, ILACE_MODE_TOP_FIRST },
+	{ "1440x1080 25i",	1440,1080, 25.,	   1, ILACE_MODE_TOP_FIRST },
+	{ "1440x1080 24p",	1440,1080, 24.,	   1, ILACE_MODE_NOTINTERLACED },
+	{ "1440x1080 23.976p",	1440,1080, 23.976, 1, ILACE_MODE_NOTINTERLACED },
 	{ "1280x720  59.94p",	1280,720,  59.94,  1, ILACE_MODE_NOTINTERLACED },
 	{ "1280x720  50p",	1280,720,  50.,    1, ILACE_MODE_NOTINTERLACED },
 	{ "1280x720  29.97p*",  1280,720,  29.97,  1, ILACE_MODE_NOTINTERLACED },
-- 
2.35.8

